docs/components/swagger.tsx

```diff
@@ -0,0 +1,204 @@
+import { Tag } from "@blueprintjs/core";
+import { PageLinks } from "df/docs/components/page_links";
+import * as styles from "df/docs/components/swagger.css";
+import React from "react";
+import {
+  Operation as IOperation,
+  Parameter as IParameter,
+  Schema,
+  Spec
+} from "swagger-schema-official";
+
+interface IProps {
+  apiHost: string;
+  spec: Spec;
+}
+
+interface IOperationWithPath extends IOperation {
+  path: string;
+  method: string;
+}
+
+export class Swagger extends React.Component<IProps> {
+  public render() {
+    const { spec, apiHost } = this.props;
+    const allOperations: IOperationWithPath[] = Object.keys(spec.paths)
+      .map(path => {
+        const schema = spec.paths[path];
+        return [
+          { ...schema.post, method: "post" },
+          { ...schema.get, method: "get" },
+          { ...schema.put, method: "put" },
+          { ...schema.delete, method: "delete" }
+        ]
+          .filter(operation => !!operation.operationId)
+          .map(operation => ({ path, ...operation }));
+      })
+      .reduce((acc, curr) => [...acc, ...curr], []);
+
+    return (
+      <div className={styles.container}>
+        <div className={styles.sidebar} />
+        <div className={styles.mainContent}>
+          <div className={styles.titleBlock}>
+            <h1>Dataform Web API</h1>
+            {allOperations.map((operation, i) => (
+              <Operation key={i} apiHost={apiHost} operation={operation} />
```

Comment:
[Don't use indexes as keys](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318)

Reply:
As a general rule yes - however this content is completely static, it never changes as the data model will never change unless the user refreshes, at which point it doesn't matter.

Reply:
OK OK I changed them